### PR TITLE
Show toast after Relocated

### LINKED ISSUE #990: On devices with sdcard, external app crashes when clicked on OPEN snackbar action
dut: Samsung

The file is being moved to sdcard, thus the crash :/

### PR DESCRIPTION
**Pull request by mTwTm**
Fixes #990

### CODE REVIEW

File: app/src/main/java/org/mozilla/focus/activity/MainActivity.java [hunk 1]
```diff
@@ -784,8 +779,8 @@ private void asyncCheckStorage() {
         Settings.getInstance(this).setRemovableStorageStateOnCreate(exist);
     }
 
-    private void showOpenSnackBar(Long rowId) {
-        DownloadInfoManager.getInstance().queryByRowId(rowId, new DownloadInfoManager.AsyncQueryListener() {
+    private void showOpenSnackBar(Long id) {
+        DownloadInfoManager.getInstance().queryByDownloadId(id, new DownloadInfoManager.AsyncQueryListener() {
             @Override
```
Comment (WeiWW) [inline app/src/main/java/org/mozilla/focus/activity/MainActivity.java]: id -> downloadId , avoid input wrong.

File: app/src/main/java/org/mozilla/focus/components/RelocateService.java [hunk 2]
```diff
@@ -215,7 +222,16 @@ private void broadcastNoPermission(final long downloadId,
         broadcastIntent.putExtra(Constants.EXTRA_FILE_PATH, srcFile.getAbsoluteFile());
         broadcastIntent.setType(mediaType);
 
-        sendBroadcast(broadcastIntent);
+        LocalBroadcastManager.getInstance(this).sendBroadcast(broadcastIntent);
         Log.d(TAG, "no permission for file relocating, send broadcast to grant permission");
     }
+
+    private void broadcastRelocateFinished(long id) {
+        final Intent broadcastIntent = new Intent(Constants.ACTION_NOTIFY_RELOCATE_FINISH);
```
Comment (WeiWW) [inline app/src/main/java/org/mozilla/focus/components/RelocateService.java]: id -> downloadId, avoid input wrong

File: app/src/main/java/org/mozilla/focus/download/DownloadInfoManager.java [hunk 3]
```diff
@@ -249,6 +249,7 @@ public void onQueryComplete(List downloadInfoList) {
                 }
             }
         });
+        return newId;
```
Comment (WeiWW) [inline app/src/main/java/org/mozilla/focus/download/DownloadInfoManager.java]: check this is rowId or downloadId, then rename it

Comment (mTwTm) [reply app/src/main/java/org/mozilla/focus/download/DownloadInfoManager.java]: This should be explicit already in
final long oldId = downloadId;
and
final long newId = manager.addCompletedDownload(

Comment (WeiWW) [reply app/src/main/java/org/mozilla/focus/download/DownloadInfoManager.java]: Ok

**Pull request merged**